src/ProjectTemplates/Web.ProjectTemplates/content/BlazorWeb-CSharp/BlazorWeb-CSharp/Components/Identity/ShowRecoveryCodes.razor

```diff
@@ -0,0 +1,32 @@
+﻿<StatusMessage Message="@StatusMessage" />
+<h3>Recovery codes</h3>
+<div class="alert alert-warning" role="alert">
+    <p>
+        <strong>Put these codes in a safe place.</strong>
+    </p>
+    <p>
+        If you lose your device and don't have the recovery codes you will lose access to your account.
+    </p>
+</div>
+<div class="row">
+    <div class="col-md-12">
+        @for (var row = 0; row < RecoveryCodes.Length; row += 2)
+        {
+            <code class="recovery-code">@RecoveryCodes[row]</code>
+
+            <text>&nbsp;</text>
+
+            <code class="recovery-code">@RecoveryCodes[row + 1]</code>
+
+            <br />
+        }
+    </div>
+</div>
+
+@code {
+    [Parameter]
+    public string[] RecoveryCodes { get; set; } = default!;
+
+    [Parameter]
+    public string StatusMessage { get; set; } = default!;
```

Comment:
If you're OK with `required` instead of ` = default!;` maybe switch over

Reply:
https://github.com/dotnet/aspnetcore/pull/50722#discussion_r1334601765

Reply:
(I'm not strongly opinionated on this, so I think it's fine to use `required` if you think it is!)